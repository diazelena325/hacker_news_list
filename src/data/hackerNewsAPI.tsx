export const mainUrl = 'https://hacker-news.firebaseio.com/v0/';
export const topStoriesUrl = `${mainUrl}topstories.json`;
export const listingUrl = `${mainUrl}item/`;

//API call to get list of listing Ids of top stories
export async function getListingIds() {
  const result: number[] = await fetch(topStoriesUrl)
    .then(response => response.json())
    .then(data => data);

  return result;
}

//API call to get story details for each listing Id
export async function getEachStory(nsId : number) {
  const result :
    {
      by: string;
      descendants: number;
      id: number;
      kids: number[];
      score: number;
      time: number;
      title: string;
      type: string;
      url: string;
    }
    = await fetch(`${listingUrl + nsId}.json`)
      .then(response => response.json())
      .then((data) => data);

  return result;
}
