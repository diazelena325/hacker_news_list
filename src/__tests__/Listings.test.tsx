import { cleanup } from '@testing-library/react';
import { getListingIds } from '../data/hackerNewsAPI';
import { storyIds } from '../fixtures/fixtures';

beforeEach(() => {
    cleanup();
    jest.resetAllMocks();
});

jest.mock('../data/hackerNewsAPI', () => ({
    getListingIds: jest.fn(),
}));
const mockGetListingIds = getListingIds as jest.MockedFunction < typeof getListingIds>;

test('rendering listing ids component with content', async () => {
    mockGetListingIds.mockResolvedValue(storyIds);

    const getteststory = await getListingIds();
    expect(getListingIds).toHaveBeenCalledTimes(1);
    expect(getteststory).toMatchSnapshot();
});