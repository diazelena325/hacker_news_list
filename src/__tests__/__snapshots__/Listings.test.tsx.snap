// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering listing ids component with content 1`] = `
Array [
  1,
]
`;
